### ISSUE TITLE
Treesitter integration on hover actions

### ISSUE THREAD

**Issue by NicolasGB** [mrcjkb/rustaceanvim#20]
### Neovim version (nvim -v)

v0.9.4

### Operating system/version

Archlinux

### Output of `:checkhealth ferris`

Everything prints OK

### How to reproduce the issue

First of all thank you for forking rust-tools, your approach to plugins is really nice, hopefully this fork will live long!

I found the repo following the bug where multiple instances of RA were spawned. Thanks for fixing that so fast!

I found a "bug" in the Rust HoverActions, previously on rust-tools, the syntax highlighting with themes and TreeSitter (correct me if I'm wrong) was used. Now when I hover functions the text seems plain and it's not parsed by the theme which uses TS.

Would be nice to maintain that as it's clearer when reading rust-docs.

Let me know if this is intended and there should be flagged as enhancement instead of bug.

### Expected behaviour

![image](https://github.com/mrcjkb/rustaceanvim/assets/17547539/b93fe984-3dcc-44b1-837e-ea8efc415252)

### Actual behaviour

![image](https://github.com/mrcjkb/rustaceanvim/assets/17547539/76cd4d92-8e04-44e3-bfcd-c08556d6371e)

### The minimal config used to reproduce this issue.

Open a rust project and hover a documented function such as tracing_subscriber

**Comment by mrcjkb** [comment]
Hmm, that's strange - it works on my end (I know that's not a very helpful comment :smile:)

Your first screen shot looks like it's Neovim's built-in hover UI (there are no actions).
I did add a new config (that defaults to `true`):

```lua
vim.g.rustaceanvim = {
  tools = {
    hover_actions = {
      replace_builtin_hover = true,
    },
  },
}
```

I guess if you set that to `false`, you'll have the old behaviour back.
But I did configure hover actions to use tree-sitter highlighting (they didn't before, iirc).
I do this [by setting `vim.bo[bufnr].ft = 'markdown'`](https://github.com/mrcjkb/rustaceanvim/blob/f68c89245c5f46753050ada35d00c38ec4e582ef/lua/rustaceanvim/hover_actions.lua#L91). But I've only tested this with Neovim nightly. Maybe it doesn't work in Neovim 0.9.4?

I'll have to try and reproduce it...
